### PR TITLE
build, macOS: Allow compiling v8 on macOS 10.15 to 13.0

### PR DESCRIPTION
**Pull request by dio**
Commit Message: build, macOS: Allow compiling on macOS 10.15 to 13.0
Additional Description: https://bugs.chromium.org/p/v8/issues/detail?id=13428. Will submit an issue to track the patch removal once this is merged.
Risk Level: Low (affects only macOS)
Testing: Manual on macOS 10.15 (Xcode 12.4), 12.6 (Xcode 14.0.1), and 13.0 (Xcode 14.1), 
Fixes https://github.com/envoyproxy/envoy/issues/23359

Signed-off-by: Dhi Aurrahman <dio@rockybars.com>

### CODE REVIEW

File: bazel/v8.patch [hunk 1]
```diff
@@ -41,14 +42,26 @@ index e957c0fad3..eee285ab60 100644
...
++                # The clang available on macOS catalina has a warning that isn't clean on v8 code.
++                "-Wno-range-loop-analysis",
++
++                # To supress warning on deprecated declaration on v8 code. For example:
++                # external/v8/src/base/platform/platform-darwin.cc:56:22: 'getsectdatafromheader_64'
++                # is deprecated: first deprecated in macOS 13.0.
++                # https://bugs.chromium.org/p/v8/issues/detail?id=13428.
++                "-Wno-deprecated-declarations",
++            ],
 +            "//conditions:default": [],
```
Comment (dio) [inline bazel/v8.patch:65]: The diff section above looks a bit confusing, but here is the full-diff of that "section".

```diff
diff --git a/bazel/defs.bzl b/bazel/defs.bzl
index e957c0fad3..5870a914aa 100644
--- a/bazel/defs.bzl
+++ b/bazel/defs.bzl
@@ -116,6 +116,7 @@ def _default_args():
         }) + select({
             "@v8//bazel/config:is_clang": [
                 "-Wno-invalid-offsetof",
+                "-Wno-unneeded-internal-declaration",
                 "-std=c++17",
             ],
             "@v8//bazel/config:is_gcc": [
@@ -151,6 +152,23 @@ def _default_args():
                 "-fno-integrated-as",
             ],
             "//conditions:default": [],
+        }) + select({
+            "@envoy//bazel:no_debug_info": [
+                "-g0",
+            ],
+            "//conditions:default": [],
+        }) + select({
+            "@v8//bazel/config:is_macos": [
+                # The clang available on macOS catalina has a warning that isn't clean on v8 code.
+                "-Wno-range-loop-analysis",
+
+                # To supress warning on deprecated declaration on v8 code. For example:
+                # external/v8/src/base/platform/platform-darwin.cc:56:22: 'getsectdatafromheader_64'
+                # is deprecated: first deprecated in macOS 13.0.
+                # https://bugs.chromium.org/p/v8/issues/detail?id=13428.
+                "-Wno-deprecated-declarations",
+            ],
+            "//conditions:default": [],
         }),
         includes = ["include"],
         linkopts = select({
```

File: bazel/v8.patch [hunk 2]
```diff
@@ -1,7 +1,8 @@
 # 1. Use already imported python dependencies
 # 2. Disable pointer compression (limits the maximum number of WasmVMs).
-# 3. Add support for --define=no_debug_info=1.
-# 4. Don't expose Wasm C API (only Wasm C++ API).
+# 3. Add support for --define=no_debug_info=1. 
+# 4. Allow compiling v8 on macOS 10.15 to 13.0.
```
Comment (moderation) [inline bazel/v8.patch:4]: Would be good to add a todo here for tracking
`TODO(dio) - remove patch when https://bugs.chromium.org/p/v8/issues/detail?id=13428 lands`
or something similar

**Review by moderation — CHANGES_REQUESTED** [review]

File: bazel/v8.patch (hunk 2 above)
Comment (dio) [reply bazel/v8.patch]: Thanks for the review, @moderation! Sure. Done in abe0ac0.

**Comment by moderation** [comment]
/lgtm deps

Thanks

**Review by keith — APPROVED** [review]

**Pull request merged**